src/main/java/core/mate/academy/impl/ExcavatorProducer.java

```diff
@@ -0,0 +1,19 @@
+package core.mate.academy.impl;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    private List<Excavator> list = new ArrayList<>();
+
+    @Override
+    public List<Excavator> get() {
```

Comment:
There seems to be a potential issue with the 'get()' method. Each time it is called, it adds new Excavators to the same list, which can lead to unexpected behavior. Instead, consider initializing a new list inside the 'get()' method and returning it.